site/cds_rdm/legacy/redirector.py

```diff
@@ -59,19 +61,72 @@ def legacy_files_redirect(legacy_id, filename):
     return redirect(url_path)
 
 
+def legacy_collection_redirect(collection_name):
+    """Redirection for legacy collections."""
+    cds_community_uuid = current_app.config["CDS_REDIRECTION_COLLECTIONS_MAPPING"].get(
+        collection_name, None
+    )
+    if not cds_community_uuid:
+        raise NoResultFound
+    community = get_community_by_uuid(cds_community_uuid)
+    query_params = request.args.copy()
+    query_params["q"] = query_params.pop("p", None)
+    if query_params["q"]:
+        url_path = url_for(
+            "invenio_app_rdm_communities.communities_detail",
+            pid_value=community.data["slug"],
+            **query_params,
+        )
+    else:
+        url_path = url_for(
+            "invenio_app_rdm_communities.communities_home",
+            pid_value=community.data["slug"],
```

Comment:
did you check if we can resolve this url (and the one above) using community uuid? we would be able to avoid creating a new method `get_community_by_uuid` and using system identity in there